File: internal/src/main/java/dev/mvmo/sicklang/internal/object/BuiltinFunctionObject.java
package dev.mvmo.sicklang.internal.object;

import dev.mvmo.sicklang.internal.builtin.BuiltinFunction;

public record BuiltinFunctionObject(BuiltinFunction function) implements SickObject {

    @Override
    public String inspect() {
        return "builtin function";
    }

    @Override
    public ObjectType objectType() {
        return ObjectType.BUILTIN;
    }

}

File: internal/src/main/java/dev/mvmo/sicklang/internal/object/BuiltinFunctionObject.kt
package dev.mvmo.sicklang.internal.`object`

import dev.mvmo.sicklang.internal.builtin.BuiltinFunction

open class BuiltinFunctionObject(val name: String, val function: BuiltinFunction) : SickObject {

    override fun inspect() =
        name

    override fun objectType() =
        ObjectType.BUILTIN

}